ModularTegustation/tegu_mobs/apocalypse_bird.dm

```diff
@@ -152,13 +180,106 @@
 	visible_message("<span class='danger'>[src] suddenly disappears!</span>")
 	forceMove(T)
 	visible_message("<span class='danger'>[src] suddenly appears in front of you!</span>")
-	SLEEP_CHECK_DEATH(3 SECONDS)
+	SLEEP_CHECK_DEATH(2 SECONDS)
+	teleport_cooldown = world.time + teleport_cooldown_time
+	attacking = FALSE
+
+/mob/living/simple_animal/hostile/megafauna/apocalypse_bird/proc/Bite()
+	if(attacking || !bite_possible)
+		return
+	attacking = TRUE
+	var/list/candidates = list()
+	for(var/mob/living/L in range(16, src))
+		if(faction_check_mob(L, FALSE))
+			continue
+		if(L.stat == DEAD)
+			continue
+		candidates += L
+	var/dir_to_target = pick(GLOB.cardinals)
+	if(LAZYLEN(candidates))
+		dir_to_target = get_cardinal_dir(get_turf(src), get_turf(pick(candidates)))
+	var/turf/source_turf = get_turf(src)
+	var/turf/area_of_effect = list()
+	var/turf/upper_bound
+	var/turf/lower_bound
+	switch(dir_to_target)
+		if(EAST)
+			for(var/j = 1 to bite_length)
+				source_turf = get_step(source_turf, EAST)
+				upper_bound = source_turf
+				lower_bound = source_turf
+				for (var/k = 1 to bite_width)
+					upper_bound = get_step(upper_bound, NORTH)
+				for (var/l = 1 to bite_width)
+					lower_bound = get_step(lower_bound, SOUTH)
+				for(var/turf/T in getline(upper_bound, lower_bound))
+					area_of_effect += T
+					new /obj/effect/temp_visual/cult/sparks(T)
+		if(WEST)
+			for(var/j = 1 to bite_length)
+				source_turf = get_step(source_turf, WEST)
+				upper_bound = source_turf
+				lower_bound = source_turf
+				for (var/k = 1 to bite_width)
+					upper_bound = get_step(upper_bound, NORTH)
+				for (var/l = 1 to bite_width)
+					lower_bound = get_step(lower_bound, SOUTH)
+				for(var/turf/T in getline(upper_bound, lower_bound))
+					area_of_effect += T
+					new /obj/effect/temp_visual/cult/sparks(T)
+		if(SOUTH)
+			for(var/j = 1 to bite_length)
+				source_turf = get_step(source_turf, SOUTH)
+				upper_bound = source_turf
+				lower_bound = source_turf
+				for (var/k = 1 to bite_width)
+					upper_bound = get_step(upper_bound, EAST)
+				for (var/l = 1 to bite_width)
+					lower_bound = get_step(lower_bound, WEST)
+				for(var/turf/T in getline(upper_bound, lower_bound))
+					area_of_effect += T
+					new /obj/effect/temp_visual/cult/sparks(T)
+		if(NORTH)
+			for(var/j = 1 to bite_length)
+				source_turf = get_step(source_turf, NORTH)
+				upper_bound = source_turf
+				lower_bound = source_turf
+				for (var/k = 1 to bite_width)
+					upper_bound = get_step(upper_bound, EAST)
+				for (var/l = 1 to bite_width)
+					lower_bound = get_step(lower_bound, WEST)
+				for(var/turf/T in getline(upper_bound, lower_bound))
+					area_of_effect += T
+					new /obj/effect/temp_visual/cult/sparks(T)
+		else
+			return
+	playsound(src, 'sound/abnormalities/apocalypse/pre_attack.ogg', 125, FALSE, 4)
+	SLEEP_CHECK_DEATH(2.5 SECONDS)
+	for(var/turf/TF in area_of_effect)
+		new /obj/effect/temp_visual/cult/sparks(TF)
```

Comment:
This seems to declare the AoE twice, no? Since sparks are already placed in the line's creation.

Reply:
probably for the best